lib/resty/websocket/client.lua

```diff
@@ -135,23 +141,30 @@ function _M.connect(self, uri, opts)
                 return nil, "custom headers must be a table"
             end
         end
+
+        host = opts.host
+
+        ssl_server_name = opts.ssl_server_name
+
+        key = opts.key
+
+        keep_response = opts.keep_response
```

Comment:
We should check the type of all of these input options when they must be strings and throw appropriate errors.

Reply:
Done! Please take a look and let me know if the returned error strings need fine-tuning.